store/storetest/channel_member_history_store.go

```diff
@@ -132,11 +141,97 @@ func testGetUsersInChannelAt(t *testing.T, ss store.Store) {
 	assert.Equal(t, joinTime, channelMembers[0].JoinTime)
 	assert.Equal(t, leaveTime, *channelMembers[0].LeaveTime)
 
-	// case 6: start and end after leave time
+	// case 6: user has joined and left the channel long before the export period begins
 	channelMembers = store.Must(ss.ChannelMemberHistory().GetUsersInChannelDuring(leaveTime+100, leaveTime+200, channel.Id)).([]*model.ChannelMemberHistory)
 	assert.Len(t, channelMembers, 0)
 }
 
+func testGetUsersInChannelAtChannelMembers(t *testing.T, ss store.Store) {
+	// create a test channel
+	channel := model.Channel{
+		TeamId:      model.NewId(),
+		DisplayName: "Display " + model.NewId(),
+		Name:        "zz" + model.NewId() + "b",
+		Type:        model.CHANNEL_OPEN,
+	}
+	channel = *store.Must(ss.Channel().Save(&channel, -1)).(*model.Channel)
+
+	// and a test user
+	user := model.User{
+		Email:    model.NewId() + "@mattermost.com",
+		Nickname: model.NewId(),
+	}
+	user = *store.Must(ss.User().Save(&user)).(*model.User)
+
+	// in this test, we're pretending that Message Export was not activated during the export period, so there's no data
+	// available in the ChannelMemberHistory table. Instead, we'll fall back to the ChannelMembers table for a rough approximation
+	// Note: This will break if another test inserts records into ChannelMemberHistory with a join time before 1000
```

Comment:
I'm not crazy about this, but I couldn't figure out a good way to wipe the contents of the ChannelMemberHistory table before running this test. I've put the time far enough in the past that it should never be a problem, but it is more brittle than I'd like it to be. If we wiped and re-created database tables before running the test suite, that would be better.

Reply:
How about just putting something like `ss.ChannelMemberHistory().GetMaster().Exec("TRUNCATE TABLE ChannelMemberHistory")` in here?

Reply:
Fixed in https://github.com/mattermost/mattermost-server/pull/7951/commits/b8a1980fff02154a005ab2a378390a912a2fe66c
I had to add `GetDbMaster()` to the interface, which seems like it could cause a leaky abstraction, but I agree that it's better than the relatively brittle test that we had before.